data/pigui/modules/station-view/04-shipMarket.lua

```diff
@@ -159,15 +159,143 @@ local function buyShip (mkt, sos)
 	mkt.popup:open()
 end
 
-local yes_no = function (binary)
+local FormatAndCompareShips = {}
+
+function FormatAndCompareShips:beginTable()
+	if not ui.beginTable("specs", 6, {"NoPadInnerX"} ) then
+		return false
+	end
+
+	ui.tableSetupColumn("name1")
+	ui.tableSetupColumn("body1")
+	ui.tableSetupColumn("indicator1", {"WidthFixed"})
+	ui.tableSetupColumn("name2")
+	ui.tableSetupColumn("body2")
+	ui.tableSetupColumn("indicator2", {"WidthFixed"})
+
+	return true
+end
+
+function FormatAndCompareShips:compare_and_draw_column(desc, str, a, b)
+	compare = a-b
+	color = styleColors.white
+	indicator = " "
+	if compare < 0 then
+		color = styleColors.warning_300
+		-- note a preceeding 3 per en space as no column padding
+		-- this is unicode point 2004
+		indicator = " ↓"
+	elseif compare > 0 then
+		color = styleColors.success_300
+		-- note a preceeding 3 per en space as no column padding
+		-- this is unicode point 2004
+		indicator = " ↑"
+	end
+
+	if self.column == 0 then
+		ui.tableNextRow()
+	end
+
+	ui.tableSetColumnIndex(0 + self.column)
+	ui.textColored(styleColors.gray_300, desc)
+
+	ui.tableSetColumnIndex(1 + self.column)
+	ui.textAlignedColored(str, 1.0, color)
+
+	ui.tableSetColumnIndex(2 + self.column)
+
+	if self.column == 0 then
+		-- this is an em space (unicode point 2003)
+		-- as we don't have and column padding
+		ui.textColored(color,indicator .. " ")
+		self.column = 3
+	else
+		ui.textColored(color,indicator)
+		self.column = 0
+	end
+end
```

Comment:
Happy to change over and use an icon if that's what you'd prefer. 

The problem I have with using the cell padding when defining the table is that it doesn't support different padding per column. Essentially here we've got 2 tables, the first with columns, each with a table with 3 columns inside and I want different cell padding for both. 

Right now we're simulating this with one table with 6 columns. Can we embed one table inside another as this would solve my problem much more neatly and I should then be able to specify the desired table padding separately for each of the tables.  It would though necessitate a different, less natural draw order as I think we'd have to draw the left column in it's entirety first and then the right.

For aesthetics, I want there to be a gap between the two 'major' columns but also want a smaller gap, between the stat and the indicator arrow.  I settled on an em space (a typographic space that is always the same width as the height of the body of a line of text) as the width between the major columns, and 1/3 of the width of the em space (so the 3 per em space) for the gap between the stat and the indicator arrow- and messed up the comment above, where I said en space for that.

I also don't really like using typographic spaces in source code like the above, I'd much rather use something like "UTF.char(2004)" to show exactly what has been chosen, as in the source editor it all just looks like whitespace, however our version of lua doesn't seem to support that and I didn't fancy writing a c util function to enable it - which would be better.

Reply:
Never mind, I just worked out how to do the columns and have them aesthetically pleasing without all the spacing stuff.

Just add an empty column in between the 2 major ones.

I'll get onto this :-)

Reply:
Done :-)